shared_model/interfaces/iroha_internal/transaction_batch_parser_impl.cpp

```diff
@@ -0,0 +1,66 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "interfaces/iroha_internal/transaction_batch_parser_impl.hpp"
+
+#include <boost/range/adaptor/indirected.hpp>
+#include <boost/range/combine.hpp>
+#include "interfaces/iroha_internal/batch_meta.hpp"
+#include "interfaces/transaction.hpp"
+
+namespace {
+  /**
+   * Zips in_range and out_range, where in_range elements are objects, parses
+   * batches based on batchMeta values of in_range, and returns a collection of
+   * corresponding sub-ranges of out_range
+   */
+  template <typename InRange, typename OutRange>
+  auto parseBatchesImpl(InRange in_range, const OutRange &out_range) {
+    std::vector<OutRange> result;
+    auto meta = [](const auto &tx) { return boost::get<0>(tx).batchMeta(); };
+    auto has_meta = [&](const auto &tx) { return static_cast<bool>(meta(tx)); };
```

Comment:
Maybe `[&](const auto &tx) -> bool {` will work? More readable than a static_case

Reply:
This error was fixed previously https://github.com/hyperledger/iroha/pull/1750/commits/3aad9567eaf9dbe2ca5c9d579024db9417b419f7
```
error: no viable conversion from 'boost::optional<std::shared_ptr<BatchMeta> >' to 'bool'

             begin_has_meta = boost::get<0>(*begin).batchMeta();

             ^                ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```